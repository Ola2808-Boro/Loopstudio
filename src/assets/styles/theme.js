export const theme = {
  colors: {
    white: "#ffffff",
    black: "	#000000",
    red: "#FF0000",
    darkGray: "	#8c8c8c",
    veryDarkGray: "#696969",
    vividMulberry: "#C505EB",
    lavenderIndigo: "#8647F8",
  },
  fontSizes: {
    s: "15px",
    m: "20px",
    l: "25px",
    xl: "30px",
  },
};
