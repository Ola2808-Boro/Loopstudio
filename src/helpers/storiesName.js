export const stories = {
  atoms: "Components/atoms/",
  molecules: "Components/molecules/",
  organisms: "Components/organisms/",
  temapltes: "Components/temapltes/",
};
