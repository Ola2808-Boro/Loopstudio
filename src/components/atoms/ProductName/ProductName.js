import React from "react";
import { StyledProductName } from "./ProductName.styles";
import PropTypes from "prop-types";
const ProductName = ({ productName = "Product name", details, color }) => {
  return (
    <StyledProductName details={details} color={color}>
      {productName}
    </StyledProductName>
  );
};

ProductName.propTypes = {
  /**Text */
  productName: PropTypes.string.isRequired,
  /**Size of product name */
  details: PropTypes.bool.isRequired,
  /**Color of product name */
  color: PropTypes.string.isRequired,
};
export default ProductName;
