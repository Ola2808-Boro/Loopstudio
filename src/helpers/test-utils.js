import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "assets/styles/theme";
import GlobalStyle from "assets/styles/GlobalStyle";
import DataProvider from "providers/DataProvider";
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <DataProvider>
        <GlobalStyle />
        {children}
      </DataProvider>
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
