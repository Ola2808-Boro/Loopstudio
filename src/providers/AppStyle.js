import React from "react";
import { ThemeProvider } from "styled-components";
import { theme } from "assets/styles/theme";
import GlobalStyle from "assets/styles/GlobalStyle";
import "assets/styles/fonts.css";

const AppStyle = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  );
};

export default AppStyle;
