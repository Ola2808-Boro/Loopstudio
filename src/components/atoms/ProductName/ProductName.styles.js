import styled from "styled-components";
import { device } from "helpers/breakpoints";

export const StyledProductName = styled.p`
  color: ${(props) =>
    props.color ? props.color : ({ theme }) => theme.colors.black};
  font-family: "Josefin Sans", sans-serif;
  letter-spacing: 1.5px;
  text-transform: uppercase;
  line-height: 1.2em;
  padding: 10px 10px;
  font-size: ${(props) =>
    props.details
      ? ({ theme }) => theme.fontSizes.l
      : ({ theme }) => theme.fontSizes.m};

  font-weight: bold;
`;
