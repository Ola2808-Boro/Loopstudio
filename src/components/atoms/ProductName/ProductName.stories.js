import ProductName from "./ProductName";
import { stories } from "helpers/storiesName";

export default {
  title: `${stories.atoms}ProductName`,
  component: ProductName,
};

const Template = (args) => <ProductName {...args} />;

export const Default = Template.bind({});

Default.args = {
  color: "black",
  productName: "Product Name",
};
