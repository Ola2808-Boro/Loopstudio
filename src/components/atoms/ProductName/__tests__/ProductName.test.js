import { render, screen } from "helpers/test-utils";
import renderer from "react-test-renderer";
import ProductName from "components/atoms/ProductName/ProductName";
import AppStyle from "providers/AppStyle";
import { StyledProductName } from "components/atoms/ProductName/ProductName.styles";
describe("Test Product Name component", () => {
  it("default style", () => {
    const tree = renderer
      .create(
        <AppStyle>
          <StyledProductName />
        </AppStyle>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("test render component", async () => {
    render(<ProductName productName={"Computer"} details={true} />);
    const element = screen.getByText("Computer");
    expect(element).toBeInTheDocument();
    expect(element).not.toHaveStyle("font-size: 25px");
  });
});
