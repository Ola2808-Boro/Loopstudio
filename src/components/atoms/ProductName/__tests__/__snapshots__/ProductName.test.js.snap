// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Product Name component default style 1`] = `
<p
  className="sc-bczRLJ kedjXS"
/>
`;
